File: Problems/04 Median of Two Sorted Arrays/Solution.js
// https://leetcode.com/problems/median-of-two-sorted-arrays/

/*
Runtime: 110 ms, faster than 92.44% of JavaScript online submissions for Median of Two Sorted Arrays.
Memory Usage: 48.1 MB, less than 40.44% of JavaScript online submissions for Median of Two Sorted Arrays.
Next challenges:
*/

/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function (nums1, nums2) {
    let arr = [...nums1, ...nums2];
    arr.sort((a, b) => a - b);
    if (arr.length % 2 == 0) {
        return (arr[(arr.length / 2) - 1] + arr[arr.length / 2]) / 2;
    } else {
        return arr[(arr.length - 1) / 2]; //give 92%
        return arr[Math.ceil(arr.length / 2) - 1];
    }
};

console.log(findMedianSortedArrays([1, 3], [2]));
console.log(findMedianSortedArrays([1, 2], [3, 4]));

File: Problems/04 Median of Two Sorted Arrays/Solution.py
# https://leetcode.com/problems/median-of-two-sorted-arrays/

"""
Runtime: 102 ms, faster than 88.99% of Python3 online submissions for Median of Two Sorted Arrays.
Memory Usage: 14.2 MB, less than 67.77% of Python3 online submissions for Median of Two Sorted Arrays.
"""


class Solution:
    def findMedianSortedArrays(self, nums1, nums2):
        nums1 = nums1+nums2
        nums1.sort()
        if len(nums1) % 2 == 0:
            return (nums1[int((len(nums1) / 2)) - 1] + nums1[int(len(nums1) / 2)]) / 2
        return nums1[int((len(nums1)-1) / 2)]


obj = Solution()
print(obj.findMedianSortedArrays([1, 3], [2]))
print(obj.findMedianSortedArrays([1, 2], [3, 4]))
